### PR TITLE
Implement issue locking functionality

### LINKED ISSUE #56: FR: Lock issues after they are fixed
@wilhuff proposed a feature for the bot to lock issues when conversation is no longer productive.  It would probably involve the bot leaving a comment like:

> Hey everyone.  It has been {time interval} since this issue was fixed so I am locking the thread.  If you're still having problems please open a new issue (link) and we're happy to discuss.

The main to-do here is to determine the criteria for locking.  Should we just do closed for {x} time?  Should we require a special label?  A milestone?

### PR DESCRIPTION
**Pull request by samtstern**
Fixes #56 

TODO:
- [x] Unit tets
- [x] Test in staging
- [x] Add to README 

It works!:
https://github.com/samtstern/BotTest/issues/93
Screen Shot 2019-09-30 at 2 20 58 PM

### CODE REVIEW

**Comment by samtstern** [comment]
@marcbaechinger can you sanity check this and make sure it looks like what you wanted?  Then I will handle the TODOs

File: functions/src/cron.ts [hunk 1]
```diff
@@ -39,18 +39,71 @@ export class CronHandler {
...
+      const closedAtMs = new Date(closedAtStr).getTime();
+      const nowMs = new Date().getTime();
+
+      if (nowMs - closedAtMs > lockMillis) {
+        actions.push(
+          new types.GithubLockAction(
+            org,
+            name,
+            issue.number,
+            `Issue was closed at ${closedAtStr} which is more than ${lockDays} ago`
```
Comment (marcbaechinger) [inline functions/src/cron.ts]: nit: what is the purpose of this text? I think it's not used yet. For us it's fine to just lock after lockDays. We currently do not add a comment to tell people why it has been locked. I guess after lockDays that's fine anyway.

File: functions/src/cron.ts [hunk 2]
```diff
@@ -39,18 +39,71 @@ export class CronHandler {
...
+    const lockDays = issueConfig.lock_days;
+    const lockMillis = lockDays * 24 * 60 * 60 * 1000;
+
+    const actions: types.Action[] = [];
+
+    const issues = await this.gh_client.getIssuesForRepo(org, name, "closed");
+    for (const issue of issues) {
+      if (issue.closed_at === null) {
+        log.warn(
+          `Closed issue ${org}/${name}/${issue.number} has no closed_at.`
```
Comment (marcbaechinger) [inline functions/src/cron.ts]: Do you know in what cases this can happen?

File: functions/src/cron.ts (hunk 1 above)
Comment (samtstern) [reply functions/src/cron.ts]: This is the "reason" which is only displayed in the audit log, not shown on GitHub.

File: functions/src/cron.ts (hunk 2 above)
Comment (samtstern) [reply functions/src/cron.ts]: This is a "this should never happen" thing but the GitHub API is not type-safe enough for me to ignore it.  Will add a comment.

**Pull request merged**